server/pulp/plugins/util/publish_step.py

```diff
@@ -1055,12 +1055,14 @@ def process_main(self):
         # any units that are already in pulp
         units_we_already_had = set()
 
-        # for any unit that is already in pulp, save it into the repo
-        for unit_dict in self.content_query_manager.get_multiple_units_by_keys_dicts(
-                self.unit_type, self.parent.available_units, self.unit_key_fields):
-            unit = self._dict_to_unit(unit_dict)
-            self.get_conduit().save_unit(unit)
-            units_we_already_had.add(unit)
+        # mongodb throws exceptions for too big queries, so we spilt it into multiple smaller ones
+        for page in misc.paginate(self.parent.available_units, 50):
```

Comment:
Do you think we should go with something larger than 50 units per page? This might cause a lot of queries, which might also not be optimal.

I'm not sure what value I would suggest, maybe 500?

If you prefer to keep 50, that's fine with me.